codecs.go

```diff
@@ -1,63 +1,41 @@
 package rsmt2d
 
-import (
-	"errors"
-	"fmt"
-)
-
-// codecType type
-type CodecType int
+import "fmt"
 
-// Erasure codes enum:
 const (
-	// RSGF8 represents Reed-Solomon codecType with an 8-bit Finite Galois Field (2^8)
-	RSGF8 CodecType = iota
-	LeopardFF8
-	LeopardFF16
+	LeopardFF16 = "LeopardFF16"
+	LeopardFF8  = "LeopardFF8"
+	RSGF8       = "RSFG8"
 )
 
-func (c CodecType) String() string {
-	switch c {
-	case RSGF8:
-		return "RSGF8"
-	case LeopardFF8:
-		return "LeopardFF8"
-	case LeopardFF16:
-		return "LeopardFF16"
-	default:
-		return "UNSUPPORTED CODEC TYPE"
-	}
-}
-
 type Codec interface {
-	encode(data [][]byte) ([][]byte, error)
-	decode(data [][]byte) ([][]byte, error)
-	codecType() CodecType
+	Encode(data [][]byte) ([][]byte, error)
+	Decode(data [][]byte) ([][]byte, error)
+	New() Codec
 	// maxChunks returns the max. number of chunks each code supports in a 2D square.
 	maxChunks() int
 }
```

Comment:
the extra `New` method is also not needed, but it will useful in the future should we be using mutlithreaded encoding/repairing

Reply:
Just out of curiosity: Why are the `Encode`/`Decode` methods exported now? 

> the extra New method is also not needed, but it will useful in the future should we be using mutlithreaded encoding/repairing

Can we introduce it when we need it instead?

Reply:
The `Encode` and `Decode` methods are exported because the functions with the same names have been removed.

Reply:
Yeah, just saw that https://github.com/lazyledger/rsmt2d/pull/46#discussion_r614427643
Thanks for clarifying.